File: leetcode/widest-vertical-area-between-two-points-containing-no-points.java
// https://leetcode.com/problems/widest-vertical-area-between-two-points-containing-no-points/

import java.util.Arrays;

class Solution {
    public int maxWidthOfVerticalArea(int[][] points) {
        Arrays.sort(points, (one, two) -> one[0] - two[0]);
        int max = 0;
        for(int i = 1; i<points.length; i++){
            if(points[i][0] - points[i-1][0] > max) max = points[i][0] - points[i-1][0];
        }
        return max;
    }
}

File: leetcode/widest_vertical_area_between_two_points_containing_no_points.py
# https://leetcode.com/problems/widest-vertical-area-between-two-points-containing-no-points


class Solution:
    def maxWidthOfVerticalArea(self, points: list[list[int]]) -> int:
        points.sort(key=lambda x: x[0])

        last: int = points[0][0]
        max_diff: int = 0
        for i, _ in points:
            max_diff = max(i - last, max_diff)
            last = i

        return max_diff
